spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -57,13 +62,18 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		this.binderFactory = binderFactory;
 	}
 
-	public Binding<MessageChannel> bindConsumer(MessageChannel inputChannel, String inputChannelName) {
+	public Collection<Binding<MessageChannel>> bindConsumer(MessageChannel inputChannel, String inputChannelName) {
 		String channelBindingTarget = this.channelBindingServiceProperties.getBindingDestination(inputChannelName);
-		Binder<MessageChannel> binder = getBinderForChannel(inputChannelName);
-		Binding<MessageChannel> binding = binder.bindConsumer(channelBindingTarget, consumerGroup(inputChannelName), inputChannel,
-				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName));
-		this.consumerBindings.put(inputChannelName, binding);
-		return binding;
+		String[] channelBindingTargets = StringUtils.commaDelimitedListToStringArray(channelBindingTarget);
+		List<Binding<MessageChannel>> bindings = new ArrayList<>();
+		for (String target : channelBindingTargets) {
+			Binder<MessageChannel> binder = getBinderForChannel(inputChannelName);
+			Binding<MessageChannel> binding = binder.bindConsumer(target, consumerGroup(inputChannelName), inputChannel,
```

Comment:
Same here. Since the `consumerGroup` and the `consumerProperties` are same for all the `destinations` for the given `inputChannelName`, these values can be obtained before iterating over the destinations to bind.